tests/test_base_plugin.py

```
@@ -202,6 +215,106 @@ def test_config_loading_without_plugin_config(self):
             self.assertEqual(plugin.response_delay, 2.5)  # DEFAULT_MESSAGE_DELAY
             self.assertEqual(plugin.channels, [])
 
+    def test_non_core_plugin_preserves_inferred_type_with_legacy_plugins_config(self):
+        """Legacy 'plugins' config should not demote inferred custom/community plugin type."""
+        legacy_config = {"plugins": {"test_plugin": {"active": True}}}
+        class_file = "/tmp/custom_plugins/test_plugin.py"  # nosec B108
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/custom_plugins/test_plugin.py"

(S108)